src/entities/explorer/hooks/use-props.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+import { getTypeArguments, getTypeDetails, checkArrayInterface } from '../lib';
+import { ITypeArguments, TypeParsedField, TypeGetTypeArgumentsReturn } from '../model';
+
+export const useExplorerProps = ({ parsedField, option }: IUseExplorerProps) => {
+  const [docs, setDocs] = useState<ITypeArguments[]>([]);
+
+  useEffect(() => {
+    let getParsedArray: TypeGetTypeArgumentsReturn;
+    if (parsedField) {
+      if (option === Options.arguments) {
+        getParsedArray = getTypeArguments(parsedField);
+      }
+      if (option === Options.details) {
+        getParsedArray = getTypeDetails(parsedField);
+      }
+      if (getParsedArray) {
+        checkArrayInterface(getParsedArray, setDocs);
+      }
+    } else setDocs([]);
```

Comment:
There are different logical `if` statements, first 2 depend on Options, third one check the value itself and the last one handle default behavior. Better combine first 2 statements with ternary operator